test/e2e/fixtures/then.go

```diff
@@ -131,6 +131,23 @@ func (t *Then) ExpectAuditEvents(filter func(event apiv1.Event) bool, blocks ...
 	return t
 }
 
+func (t *Then) ExpectWorkflowTemplates(block func(t *testing.T, templates []wfv1.Template)) *Then {
```

Comment:
The reason why this didn't exist prior is because we don't expect the templates in a workflow to ever change after submission